src/adapter/http/src/smart_protocol/messages.rs

```diff
@@ -185,12 +185,13 @@ pub struct MetadataBlocksBatch {
     pub payload: Vec<u8>,
 }
 
-/// Transfer size estimate
+/// Transfer plan
 #[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
-pub struct TransferSizeEstimate {
+pub struct TransferPlan {
     pub num_blocks: u32,
     pub num_objects: u32,
     pub num_records: u64,
+    pub new_watermark: Option<DateTime<Utc>>,
```

Comment:
I'm very much against exposing watermark on transfer protocol level - if looks out of place here.

Whoever is performing the transfer is free to analyze blocks to see if watermark changed or not, but let's not embed it into the protocol.

Reply:
I was trying to avoid the excessive re-scan.

Reply:
Reverted